src/grammar.coffee

```diff
@@ -669,16 +669,25 @@ grammar =
   ForBody: [
     o 'FOR Range',                              -> source: (LOC(2) new Value($2))
     o 'FOR Range BY Expression',                -> source: (LOC(2) new Value($2)), step: $4
-    o 'ForStart ForSource',                     -> $2.own = $1.own; $2.ownTag = $1.ownTag; $2.name = $1[0]; $2.index = $1[1]; $2
+    o 'ForStart ForSource',                     ->
+        $2[prop] = $1[prop] for prop in ['await', 'awaitTag', 'own', 'ownTag']
```

Comment:
please move this kind of code to the AST noes

Reply:
OK. That code was already there, I just added 'await' parts.
But, I agree. It belongs to the `nodes`.